Source/JavaScriptCore/dfg/DFGAbstractInterpreterInlines.h

```diff
@@ -5042,16 +5042,36 @@ bool AbstractInterpreter<AbstractStateType>::executeEffects(unsigned clobberLimi
     case GetGlobalVar: {
         if (node->hasDoubleResult())
             setTypeForNode(node, SpecBytecodeRealNumber);
-        else
+        else {
+            // Emptiness is monotonic. And GetGlobalLexicalVariable and GetGlobalVar are tied to GlobalObject.
+            // So long as it is not unlinked.
+            if (!m_graph.m_plan.isUnlinked()) {
+                JSValue concurrentlyRead = node->variablePointer()->get();
+                if (concurrentlyRead != JSValue()) {
```

Comment:
nit: `!concurrentlyRead.isEmpty()`

Reply:
Changed it to boolean check `if (JSValue oncurrentlyRead = node->variablePointer()->get())` as it returns false for jsempty (and it is the pattern we are using in JSC).